src/blockdev.rs

```diff
@@ -571,10 +578,14 @@ impl SavedPartitions {
             Ok(gpt) => gpt,
             Err(gptman::Error::InvalidSignature) => {
                 // no GPT on this disk, so no partitions to save
-                return Ok(Self {
-                    sector_size,
-                    partitions: Vec::new(),
-                });
+                if disk_has_mbr(disk) {
+                    bail!("Unsupported: Cannot save partitions from a MBR disk.")
+                } else {
```

Comment:
We've just bailed in the `true` case, so the `else` is unnecessary and adds indentation.